sdk/python/lib/pulumi/provider/server.py

```diff
@@ -112,19 +113,28 @@ async def Construct(
         await self.lock.acquire()
         try:
             return await self._construct(request, context)
-        except InputPropertiesError as e:
-            status = self.create_grpc_invalid_properties_status(e.message, e.errors)
-            await context.abort_with_status(status)
-            # We already aborted at this point
-            raise
-        except InputPropertyError as e:
-            status = self.create_grpc_invalid_properties_status(
-                "", [{"property_path": e.property_path, "reason": e.reason}]
-            )
-            await context.abort_with_status(status)
-            # We already aborted at this point
-            raise
-
+        except Exception as e:  # noqa
+            if isinstance(e, InputPropertiesError):
+                status = self.create_grpc_invalid_properties_status(e.message, e.errors)
+                await context.abort_with_status(status)
+                # We already aborted at this point
+                raise
+            elif isinstance(e, InputPropertyError):
+                status = self.create_grpc_invalid_properties_status(
+                    "", [{"property_path": e.property_path, "reason": e.reason}]
+                )
+                await context.abort_with_status(status)
+                # We already aborted at this point
+                raise
+            else:
+                stack = traceback.extract_tb(e.__traceback__)[:]
+                if len(stack) > 3:
```

Comment:
I guess this might be a bit brittle if we introduce or remove any layers of abstraction there.  More likely that we introduce more so worst case people should seem a little too much of the stack trace, which seems fine.

Reply:
The integration test makes sure the we have all the expected lines of the user's code, so if we reduce the layers, we'll catch it.

I'll update the test to expect the exact amount for the test code, that way we'll catch adding layers also.